concrete/controllers/panel/dashboard.php

```diff
@@ -82,4 +84,131 @@ public function view()
         $this->set('ui', $ui);
         $this->set('tab', $tab);
     }
+
+    public function checkForMenu($c)
+    {
+        $result = $this->renderActiveSubPanels($c);
+        if (!$result || $result == '') {
+            return null;
+        } else {
+            return 1;
+        }
+    }
+
+    public function renderActiveSubPanels($c)
+    {
+        $nh = Core::make('helper/navigation');
+        $trail = $nh->getTrailToCollection($c);
+        $systemSettings = 79;
+        $dashboard = 2;
+        $systemSettingsPageID = 80;
+        $accountID = 192;
+        $messages = 195;
+        $waiting = 186;
```

Comment:
Why are these hard-coded ID's and why not retrieve by "collectionPath" instead of "byID" later on? Hard-coding ID's is not fail-safe

Reply:
Good point.  I fixed this just now to use getCollectionID.  I still need two functions though because some pages need to generate the pageList using the parent ID of the current page while others do not.